test/Mono.Linker.Tests.Cases/DataFlow/MethodByRefParameterDataFlow.cs

```diff
@@ -58,6 +73,108 @@ class TestTypeWithRequires
 			public static void Requires () { }
 		}
 
+		static void TestReadFromRefParameter ()
+		{
+			Type typeWithMethods = null;
+			TryGetAnnotatedValue (ref typeWithMethods);
+			typeWithMethods.RequiresPublicMethods ();
+		}
+
+		static void TestReadFromOutParameter_PassedTwice ()
+		{
+			Type typeWithMethods = null;
+			TryGetAnnotatedValueFromValue (typeWithMethods, ref typeWithMethods);
+			typeWithMethods.RequiresPublicMethods ();
+		}
+
+		// https://github.com/dotnet/linker/issues/2632
+		// This test should generate a warning since there's mismatch on annotations
+		// [ExpectedWarning ("IL2062", nameof (DataFlowTypeExtensions.RequiresPublicFields))]
+		static void TestReadFromRefParameter_MismatchOnOutput ()
+		{
+			Type typeWithMethods = null;
+			TryGetAnnotatedValue (ref typeWithMethods);
+			typeWithMethods.RequiresPublicFields ();
+		}
+
+		// https://github.com/dotnet/linker/issues/2632
+		// This test should generate a warning since there's mismatch on annotations
+		// [ExpectedWarning ("IL2062", nameof (DataFlowTypeExtensions.RequiresPublicFields))]
+		static void TestReadFromRefParameter_MismatchOnOutput_PassedTwice ()
+		{
+			Type typeWithMethods = null;
+			TryGetAnnotatedValueFromValue (typeWithMethods, ref typeWithMethods);
+			typeWithMethods.RequiresPublicFields ();
+		}
+
+		[ExpectedWarning ("IL2072", nameof (TryGetAnnotatedValue))]
+		// https://github.com/dotnet/linker/issues/2632
+		// This second warning should not be generated, the value of typeWithMethods should have PublicMethods
+		// after the call with out parameter.
+		[ExpectedWarning ("IL2072", nameof (DataFlowTypeExtensions.RequiresPublicMethods))]
+		static void TestReadFromRefParameter_MismatchOnInput ()
+		{
+			Type typeWithMethods = GetTypeWithFields ();
+			TryGetAnnotatedValue (ref typeWithMethods);
+			typeWithMethods.RequiresPublicMethods ();
+		}
+
+		[ExpectedWarning ("IL2072", nameof (TryGetAnnotatedValue))]
```

Comment:
nit: TryGetAnnotatedValueFromValue to be consistent, not actually important from the standpoint of the testing infra